MonoGame.Framework/Input/Keyboard.Windows.cs

```diff
@@ -4,21 +4,57 @@
 
 using System;
 using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices;
 
 namespace Microsoft.Xna.Framework.Input
 {
     public static partial class Keyboard
     {
-        static List<Keys> _keys;
+        private static readonly Keys[] DefinedKeys;
+
+        private static readonly byte[] _keyState = new byte[256];
```

Comment:
This might need to be saved for a later PR...  but this uses 8x the memory it needs to.  We could easily encode all the state into 4 `uint64`s as bits.  It would likely be much more efficient to query as well.